compiler/lib/deadcode.ml

```diff
@@ -42,7 +42,7 @@ type t =
   { blocks : block Addr.Map.t
   ; live : variable_uses
   ; defs : def list array
-  ; mutable reachable_blocks : Addr.Set.t
+  ; reachable_blocks : BitSet.t
   ; pure_funs : Var.Set.t
```

Comment:
The PR doesn't change this code, but 
- Have you considered making pure_funs a bitset ?
- Maybe the result of `Pure_fun.f`  could be made abstract.

Reply:
I don't think it is worth it to use a bitset.